### PR TITLE
Add unit tests and bug fixes for Admin Create/Update Product pages

### PR DESCRIPTION
**Pull request by lamchenghou**
### [frontend] client/src/pages/admin/UpdateProduct.test.js

✓ fetches single product and categories on mount (94 ms)
✓ updates product successfully (78 ms)
✓ shows error toast when update fails (90 ms)
✓ sets shipping value to '0' when selecting No (84 ms)
✓ deletes product successfully (33 ms)
✓ shows error toast when delete fails (83 ms)

> 🐞 (bug): shipping field missing in update request body. This caused updates with shipping changes to silently fail.
> 🔧 (fix): Added shipping field to update request body.

> 🐞: Update branch logic was flipped, causing incorrect update payload when switching between shipping options.
> 🔧: Corrected branch condition to align with expected behavior.

### [frontend] client/src/pages/admin/CreateProduct.test.js

✓ fetches categories on render (78 ms)
✓ shows error toast if fetching categories fails (72 ms)
✓ submits product and sends FormData payload (69 ms)
✓ on failure response it should not navigate and should show error (32 ms)
✓ on success response should show success and navigate (1009 ms)
✓ shows error toast if create product throws (50 ms)
✓ shows error toast that name is required if name is not filled (34 ms)
✓ does not allow negative values for price (15 ms)
✓ does not allow negative values for quantity (15 ms)

> 🐞: Form did not enforce constraints on negative numeric inputs for price and quantity.
> 🔧: Added validation to disallow negative numbers.

> 🐞: Missing form data property in payload caused inconsistent product creation behavior.
> 🔧: Ensured FormData includes all required props before submission.

> Note: AI was used for some misc. purposes such as generation of stubs, mock data and code review.

### CODE REVIEW

File: client/src/pages/admin/UpdateProduct.test.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+    // Assert
+    await waitFor(() =>
+      expect(toast.error).toHaveBeenCalledWith(
+        expect.stringMatching(/something went wrong/i)
+      )
+    );
+  });
+
+  it("sets shipping value to '0' when selecting No", async () => {
```
Comment (lamchenghou) [inline client/src/pages/admin/UpdateProduct.test.js:161]: This is an example of a state-based test (finite state machine test generation technique), where we select `No` from the combo-box, and expect the `shipping` state to be `0`. We check this by also employing communication-based testing, where we see what `FormData` `axios.put` has been called with, and see whether `{ shipping: '0' }` was passed in the form request body.

File: client/src/pages/admin/UpdateProduct.js [hunk 2]
```diff
@@ -70,15 +72,17 @@ const UpdateProduct = () => {
       productData.append("quantity", quantity);
       photo && productData.append("photo", photo);
       productData.append("category", category);
-      const { data } = axios.put(
+      productData.append("shipping", shipping);
```
Comment (lamchenghou) [inline client/src/pages/admin/UpdateProduct.js:75]: Missing `productData['shipping']` bug is fixed here.

File: client/src/pages/admin/CreateProduct.test.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+    // Assert
+    await waitFor(() =>
+      expect(toast.error).toHaveBeenCalledWith(
+        expect.stringContaining("Name is Required")
+      )
+    );
+  });
+
+  it("does not allow negative values for price", async () => {
```
Comment (lamchenghou) [inline client/src/pages/admin/CreateProduct.test.js:192]: Boundary value analysis

File: client/src/pages/admin/CreateProduct.test.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    // Arrange
+    render(<CreateProduct />);
+
+    const priceInput = screen.getByPlaceholderText(/write a price/i);
+
+    // Act
+    fireEvent.change(priceInput, { target: { value: "-5" } });
+
+    // Assert
+    await waitFor(() => expect(priceInput.value).toBe("0"));
```
Comment (lamchenghou) [inline client/src/pages/admin/CreateProduct.test.js:202]: Example of state-based testing
